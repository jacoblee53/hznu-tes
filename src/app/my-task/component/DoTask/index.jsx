import React from 'react'
import { Card, Upload, Icon, Divider, Row, Col, Tag, Statistic } from 'antd'
import { inject, observer } from 'mobx-react'
import moment from 'moment'

import './index.less'

@inject('myTaskActions', 'myTaskStore', 'userStore')
@observer
class DoTask extends React.Component {
  constructor(props) {
    super(props)
    this.taskId = props.match && props.match.params.id
  }

  async componentDidMount() {
    this.fetchCurrentTask()
    this.fetchDotask()
  }

  fetchCurrentTask = () => {
    this.props.myTaskActions.fetchCurrentTask({
      id: this.taskId
    })
  }

  fetchDotask = () => {
    this.props.myTaskActions.fetchCurrentDotask({
      taskId: this.taskId
    })
  }

  componentWillUnmount() {
    this.props.myTaskActions.setCurrentTask({})
  }

  render() {
    const { currentDotask, currentTask } = this.props.myTaskStore
    const { taskDesc, taskName, endDate } = currentTask
    const isEnd = moment() >moment(endDate)
    const extraContent = (
      <div style={{display: 'flex'}}>
        {isEnd ?  <Tag color='#e85600'>已截止</Tag> : <Tag color='#32b643'>进行中</Tag>}
        <Statistic.Countdown className='countdown' value={endDate} format='D天H时m分' />
      </div>
    )

    return (
      <Card
        title={taskName || ''}
        bordered
        extra={extraContent}
      >
        <Divider orientation='left'>任务描述</Divider>
        <pre>{taskDesc}</pre>

        <Divider orientation='right'>任务提交</Divider>
        <Row type='flex' justify='space-around'>
          <Col span={7}>
            <Upload.Dragger
              disabled={isEnd}
              multiple={false}
            >
              <p className='ant-upload-drag-icon'>
                <Icon style={{fontSize:25}} type='file-word' />
              </p>
              <p className='ant-upload-hint'>点击提交word文档</p>
            </Upload.Dragger>
          </Col>
          <Col span={7}>
            <Upload.Dragger
              disabled={isEnd}
              multiple={false}
            >
              <p className='ant-upload-drag-icon'>
                <Icon style={{fontSize:25}} type='file-ppt' />
              </p>
              <p className='ant-upload-hint'>点击提交ppt文档</p>
            </Upload.Dragger>
          </Col>
          <Col span={7}>
            <Upload.Dragger
              disabled={isEnd}
              multiple={false}
            >
              <p className='ant-upload-drag-icon'>
                <Icon style={{fontSize:25}} type='video-camera' />
              </p>
              <p className='ant-upload-hint'>点击提交视频压缩文件</p>
            </Upload.Dragger>
          </Col>
        </Row>
      </Card>
    )
  }
}

export default DoTask