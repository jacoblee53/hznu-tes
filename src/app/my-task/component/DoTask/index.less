.ant-upload-drag-icon {
  margin-bottom: 5px!important;
}

.countdown {
  :global .ant-statistic-content {
    font-size: 16px;
  }
}