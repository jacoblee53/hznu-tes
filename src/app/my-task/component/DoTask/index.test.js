import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import injects from '../../../../inject/index'
import DoTask from './index'

let wrappedComponent = <DoTask {...injects} />

let wrapper

describe('src/app/my-task/component/index.test.js', () => {
  it('should render correctly', () => {
    wrapper = mount(wrappedComponent)
    expect(wrapper.html()).toBeTruthy()
  })

  it('should matchSnapshot', () => {
    const tree = renderer.create(wrappedComponent)
    expect(tree).toMatchSnapshot()
  })
})
