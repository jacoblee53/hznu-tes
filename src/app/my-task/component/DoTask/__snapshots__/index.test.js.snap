// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app/my-task/component/index.test.js should matchSnapshot 1`] = `
<div
  className="ant-card ant-card-bordered"
>
  <div
    className="ant-card-head"
    style={Object {}}
  >
    <div
      className="ant-card-head-wrapper"
    >
      
      <div
        className="ant-card-extra"
      >
        <div
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div
            className="ant-tag ant-tag-has-color"
            style={
              Object {
                "backgroundColor": "#32b643",
              }
            }
          >
            进行中
          </div>
          <div
            className="ant-statistic countdown"
          >
            <div
              className="ant-statistic-content"
            >
              <span
                className="ant-statistic-content-value"
              >
                0天0时0分
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-divider ant-divider-horizontal ant-divider-with-text-left"
    >
      <span
        className="ant-divider-inner-text"
      >
        任务描述
      </span>
    </div>
    <pre />
    <div
      className="ant-divider ant-divider-horizontal ant-divider-with-text-right"
    >
      <span
        className="ant-divider-inner-text"
      >
        任务提交
      </span>
    </div>
    <div
      className="ant-row-flex ant-row-flex-space-around"
    >
      <div
        className="ant-col ant-col-7"
      >
        <span
          className=""
        >
          <div
            className="ant-upload ant-upload-drag"
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
          >
            <span
              className="ant-upload ant-upload-btn"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
              onKeyDown={[Function]}
              role="button"
              style={
                Object {
                  "height": undefined,
                }
              }
              tabIndex="0"
            >
              <input
                accept=""
                directory={null}
                multiple={false}
                onChange={[Function]}
                style={
                  Object {
                    "display": "none",
                  }
                }
                type="file"
                webkitdirectory={null}
              />
              <div
                className="ant-upload-drag-container"
              >
                <p
                  className="ant-upload-drag-icon"
                >
                  <i
                    aria-label="icon: file-word"
                    className="anticon anticon-file-word"
                    style={
                      Object {
                        "fontSize": 25,
                      }
                    }
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="file-word"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0 0 42 42h216v494zM528.1 472h-32.2c-5.5 0-10.3 3.7-11.6 9.1L434.6 680l-46.1-198.7c-1.3-5.4-6.1-9.3-11.7-9.3h-35.4a12.02 12.02 0 0 0-11.6 15.1l74.2 276c1.4 5.2 6.2 8.9 11.6 8.9h32c5.4 0 10.2-3.6 11.6-8.9l52.8-197 52.8 197c1.4 5.2 6.2 8.9 11.6 8.9h31.8c5.4 0 10.2-3.6 11.6-8.9l74.4-276a12.04 12.04 0 0 0-11.6-15.1H647c-5.6 0-10.4 3.9-11.7 9.3l-45.8 199.1-49.8-199.3c-1.3-5.4-6.1-9.1-11.6-9.1z"
                      />
                    </svg>
                  </i>
                </p>
                <p
                  className="ant-upload-hint"
                >
                  点击提交word文档
                </p>
              </div>
            </span>
          </div>
          <div
            className="ant-upload-list ant-upload-list-text"
          />
        </span>
      </div>
      <div
        className="ant-col ant-col-7"
      >
        <span
          className=""
        >
          <div
            className="ant-upload ant-upload-drag"
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
          >
            <span
              className="ant-upload ant-upload-btn"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
              onKeyDown={[Function]}
              role="button"
              style={
                Object {
                  "height": undefined,
                }
              }
              tabIndex="0"
            >
              <input
                accept=""
                directory={null}
                multiple={false}
                onChange={[Function]}
                style={
                  Object {
                    "display": "none",
                  }
                }
                type="file"
                webkitdirectory={null}
              />
              <div
                className="ant-upload-drag-container"
              >
                <p
                  className="ant-upload-drag-icon"
                >
                  <i
                    aria-label="icon: file-ppt"
                    className="anticon anticon-file-ppt"
                    style={
                      Object {
                        "fontSize": 25,
                      }
                    }
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="file-ppt"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M424 476c-4.4 0-8 3.6-8 8v276c0 4.4 3.6 8 8 8h32.5c4.4 0 8-3.6 8-8v-95.5h63.3c59.4 0 96.2-38.9 96.2-94.1 0-54.5-36.3-94.3-96-94.3H424zm150.6 94.3c0 43.4-26.5 54.3-71.2 54.3h-38.9V516.2h56.2c33.8 0 53.9 19.7 53.9 54.1zm280-281.7L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0 0 42 42h216v494z"
                      />
                    </svg>
                  </i>
                </p>
                <p
                  className="ant-upload-hint"
                >
                  点击提交ppt文档
                </p>
              </div>
            </span>
          </div>
          <div
            className="ant-upload-list ant-upload-list-text"
          />
        </span>
      </div>
      <div
        className="ant-col ant-col-7"
      >
        <span
          className=""
        >
          <div
            className="ant-upload ant-upload-drag"
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
          >
            <span
              className="ant-upload ant-upload-btn"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
              onKeyDown={[Function]}
              role="button"
              style={
                Object {
                  "height": undefined,
                }
              }
              tabIndex="0"
            >
              <input
                accept=""
                directory={null}
                multiple={false}
                onChange={[Function]}
                style={
                  Object {
                    "display": "none",
                  }
                }
                type="file"
                webkitdirectory={null}
              />
              <div
                className="ant-upload-drag-container"
              >
                <p
                  className="ant-upload-drag-icon"
                >
                  <i
                    aria-label="icon: video-camera"
                    className="anticon anticon-video-camera"
                    style={
                      Object {
                        "fontSize": 25,
                      }
                    }
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="video-camera"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M912 302.3L784 376V224c0-35.3-28.7-64-64-64H128c-35.3 0-64 28.7-64 64v576c0 35.3 28.7 64 64 64h592c35.3 0 64-28.7 64-64V648l128 73.7c21.3 12.3 48-3.1 48-27.6V330c0-24.6-26.7-40-48-27.7zM712 792H136V232h576v560zm176-167l-104-59.8V458.9L888 399v226zM208 360h112c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H208c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
                      />
                    </svg>
                  </i>
                </p>
                <p
                  className="ant-upload-hint"
                >
                  点击提交视频压缩文件
                </p>
              </div>
            </span>
          </div>
          <div
            className="ant-upload-list ant-upload-list-text"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
